eo-runtime/src/main/java/org/eolang/ExprReduce.java

```diff
@@ -74,71 +67,138 @@
+     * <p/>Definition example:
+     * <code><pre>
+     *     final VarargExpr&lt;Long&gt; st = new Args<>(
+     *         Long.class,
+     *         (acc, x) -> acc / x,
+     *         x -> {
+     *             String msg = "";
+     *             if (x.equals(0.0)) {
+     *                 msg = "division by zero is infinity";
+     *             }
+     *             return msg;
+     *         }
+     *     );
+     * </pre></code>
+     * @param <T> Type of arguments passing to validation and reduction functions
+     * @since 1.0
+     */
+    public static final class Args<T> {
+
+        /**
+         * Varargs type.
+         */
+        private final Class<T> type;
+
+        /**
+         * Reduction.
+         */
+        private final BiFunction<T, T, T> reduction;
+
+        /**
+         * Validation.
+         */
+        private final Function<T, String> validation;
+
+        /**
+         * Ctor.
+         *
+         * @param type Type of varargs
+         * @param reduction Reduction operation on consecutive varags
+         * @param validation Validation function
+         */
+        public Args(
+            final Class<T> type,
+            final BiFunction<T, T, T> reduction,
+            final Function<T, String> validation
+        ) {
+            this.type = type;
+            this.reduction = reduction;
+            this.validation = validation;
+        }
+
+        /**
+         * Ctor.
+         *
+         * @param val New object that should be added
+         * @param oper Operation name
+         * @param idx Index or the currently processed argument
+         */
+        public void checkType(final Object val, final String oper, final int idx) {
+            if (!val.getClass().getCanonicalName().equals(this.type.getCanonicalName())) {
+                throw new ExFailure(
+                    "The %dth argument of '%s' is not a(n) %s: %s",
+                    idx + 1, oper, this.type.getSimpleName(), val
+                );
+            }
+        }
+
+        /**
+         * Ctor.
+         *
+         * @param rho Rho object
+         * @return Retuns the default state of accumulator
+         */
+        public T accumulator(final Phi rho) {
+            return new Param(rho).strong(this.type);
+        }
+
+        /**
+         * Ctor.
+         *
+         * @param val New object that should be added
+         * @return Returns empty string if validation passed and error message else
+         */
+        public String validate(final Object val) {
+            final T typed = this.type.cast(val);
+            return this.validation.apply(typed);
+        }
+
+        /**
+         * Ctor.
+         *
+         * @param acc Accumulator of the reduce method
+         * @param val New object that should be added
+         * @return Returns new state of accumulator after adding a new value
+         */
+        public T reduce(final T acc, final Object val) {
```

Comment:
@potatmen seems like not a function for this object.

Reply:
@mximp see previous reply